src/redux/ui/screens/data/actions.ts

```diff
@@ -223,36 +235,202 @@ export const handleCta = (
                     });
                     break;
 
+                case 'delegateToValidator': {
+                    // Run Screen Validations
+
+                    // Take amount from screen
+                    const account = getSelectedAccount(state);
+                    const chainId = getChainId(state, account.blockchain);
+
+                    const screenKey = getScreenDataKey({
+                        pubKey: getSelectedWallet(state)?.walletPublicKey,
+                        blockchain: account?.blockchain,
+                        chainId: String(chainId),
+                        address: account?.address,
+                        step: action.params?.params?.step,
+                        tab: undefined
+                    });
+
+                    // Open Process Tx
+                    if (
+                        action.params?.params?.validatorId &&
+                        action.params?.params?.validatorName &&
+                        action.params?.params?.token
+                    ) {
+                        const { token, validatorId, validatorName } = action.params.params;
+
+                        if (
+                            state.ui.screens.inputData &&
+                            state.ui.screens.inputData[screenKey]?.data?.amount
+                        ) {
+                            const amount = state.ui.screens.inputData[screenKey].data.amount;
+
+                            const validators = [buildDummyValidator(validatorId, validatorName)];
+
+                            delegate(
+                                getSelectedAccount(state),
+                                amount,
+                                validators,
+                                token,
+                                undefined, // feeOptions
+                                undefined
+                            )(dispatch, getState);
+                        }
+                    }
+                    break;
+                }
+
+                case 'switchNodeDelegateToValidator': {
```

Comment:
Let's keep consistent naming for actions, use either camel case of uppercase with underline (e.g. SINGLE_SELECTION). I'm fine either way, but keep them all in the same pattern